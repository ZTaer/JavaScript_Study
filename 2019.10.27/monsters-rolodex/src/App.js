// import 可以多类型导入
import React,{ Component } from 'react';
import logo from './logo.svg';
import './App.css';
import axios from 'axios';
import { CardList } from './components/card-list/card-list.component';

class App extends Component{
  constructor(){
   super();
    this.state = {
      monsters: [],
    }
  }

  // react-关于componentDidMount()此函数是在，componet中render()完只后执行的内容。( 等待笔记 )
    // 0. 经常用于请求数据,改变this.state数据,以至于react又要调用render()渲染界面。
  componentDidMount(){

    // async异步套装请求怪物数据
    async function getMonsters(){
      try{
        const users = await axios('https://jsonplaceholder.typicode.com/users');
        return users;
      }catch( error ){
          console.log( error )
      }
    }
    getMonsters().then( cur => {
      this.setState( { monsters: cur.data } );
    } );

  }
  
  render(){
    // map()函数渲染JSX方法使用( 用于循环数组, 只不过在这里是的内容是JSX )( 等待笔记 )
        // 0. 使用map函数的优势:
          // a) react非常的聪明, 如果map循环的数据中有变化, 那么他只会渲染发生变化的内容, 不会全部重新渲染
          // b) 这样的方法大大提高了效率 
        // 1. JSX中map函数使用注意事项: 
          // a) JSX中大括号中才能写JS语法: { JS语法 }
          // b) key属性，非常重要react用于标识不同的标签, 
            // 0. 有map()的地方一定要有key配合
            // 1. key一般存储id, 放置于JSX标签属性
          // c) 如果有cur => { return xxx; }别忘记return, map()才能重组数组;
        // 2. map()无return写法 
          // { this.state.monsters.map( cur => <h1 key={ cur.id } > { cur.name } </h1> ) }
        // 3. map()有return写法, 如下:
    console.log(this.state.monsters);
    return(
      <div className="App">
        <CardList monsters={this.state.monsters} />
      </div>
    );
  }

}

/* 1. 测试Componet/render( return(JSX) )/onClick/this.setState({})
class App extends Component {
  constructor(){
    // 继承蓝图属性
    super();

    this.state = {
      string: '这是一个测试',
    };
    this.test = {
      name: 'ztaer',
    };

  }

  // render()渲染JSX到前端( 等待笔记 )
  // 0. 函数本身为Component类中的函数, 经过蓝图继承, 可直接调用蓝图函数.
  // 1. 使用方法: 
    // a) 前提条件是在Component类下
    // b) render(){ return( JSX内容 ) };
  render(){
    return(

      // JSX是React模仿HTML的内容( 等待笔记 )
        // 0. JSX大致于HTML相似，只不过更加方便改变参数在JS中
        // 1. 注意className == class: 目的是为了区分JS的class
        // 2. JSX中大括号内才可以写JS语法: 如src={ logo }
      // JSX-关于onClick={ function }用法( 等待笔记 )
        // 0. 在JSX中添加此方法后，当用户单击时，将重新执行render()函数，也就是前端界面将重新渲染
        // 1. 经常使用此方法来改变某个变量参数, 来反应到前端
        // 如: onClick={ ()=>{ this.setState({ string: 'change this' }) } }
      // this.setState({ string: 'test' }); == this.state.string = 'test';( 等待笔记 )
        // 0. 此方法能轻松的改变对象属性参数
        // 1. 注意: 只能使用在this.state对象中, 其它名称对象不行
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <p>
            { this.state.string }
          </p>
          <button onClick={ () => { this.setState({ string: '测试onClick函数，以及this.setState({})直接访问对象属性的写法' }) } } >
            Click Change Text
          </button>
        
        </header>
      </div>
    );

  }

}
*/

/* 0. 原始内容
function App() {
  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          { this.state.string }
        </p>
        <button onClick={ () => { this.setState({ string: '测试onClick函数，以及this.setState({})直接访问对象属性的写法' }) } } >
          Click Change Text
        </button>
      
      </header>
    </div>
  );
}
*/



export default App;
