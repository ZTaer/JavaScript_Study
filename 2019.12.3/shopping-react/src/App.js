import React from 'react';
import { Switch,Route,Redirect } from 'react-router-dom';

import Header from './components/header/header.component';
import HomePage from './pages/homepage/homepage.component';
import ShopPage from './pages/shoppage/shoppage.component';
import SignPage from './pages/signpage/signpage.component';
import CheckoutPage from './pages/checkout/checkout.component';

import { auth, createUserProfileDocument } from './firebase/firebase.config';

import { connect } from 'react-redux';
import { setCurrentUser } from './redux/user/user.actions';

import { createStructuredSelector } from 'reselect';
import { selectUserCurrentUser } from './redux/user/user.selectors';

// 测试路由 - 2
const TelPage = props => ( <div> <h2 className="display-2" > { props.match.params.proName }: 等待建设页面 </h2></div> );
// 在Switch标签外不受路由影响 - 无论页面如何变化,组件依然显示存在( 完成笔记 ) 
    // 0. 把导航栏放在switch之外，这样导航栏将一直存在。不会受路由的控制 
    // 1. 这是一个非常非常重要的功能 - 不受页面刷新影响方法

class App extends React.Component {

  unsubscribeFromAuth = null; // Google登陆验证防内存泄漏( 完成笔记 )
  
  // 获取登陆用户信息( 完成笔记 )
  componentDidMount(){
    const { setCurrentUser } = this.props;

    this.unsubscribeFromAuth = auth.onAuthStateChanged( async user => {

      // 如果用户登陆
      if( user ){
        // firebase-onSnapshot()监听文档对象(快照对象)方便数据更新(  完成笔记)
          // 0. onSnapshot(props=>{xx}) 用于监听快照对象,如果数据发生变化,方便数据变化时实时更新
          // 1. props用于传递快照对象的数据,于监听快照对象无疑
        const userRef = await createUserProfileDocument( user ); // React什么周期组件内,可以使用await等待异步数据(  完成笔记)
        userRef.onSnapshot( props => {
          setCurrentUser({
            id: props.id,
            ...props.data()
          });
        } );
      }

      // 注意:此乃谷歌原始配置信息
      setCurrentUser( user );
    } );
  }

  // 卸载组件时,退出登陆( 完成笔记 )
  componentWillUnmount(){
    this.unsubscribeFromAuth(); // class中赋值骚操作,相当于this.unsubscribeFromAuth = null;( 完成笔记 )
  }

  render(){
    return(
      <div className="App">
        <Header />
        <Switch>
          <Route exact path='/' component={HomePage} />
          <Route path='/shop' component={ShopPage} />
          {
            // 关于路由render属性的功能( 完成笔记 )
              // 0. 用法: <Route exact path render={()=> <自定义标签 />} >
              // 1. 有render属性则不能有component属性
              // 2. 必需要有exact属性
            // 关于路由Redirect的使用,重定向指定页面( 完成笔记 )
              // 0. 导入Redirect: import { Switch,Route,Link,Redirect } from 'react-router-dom';
              // 1. 使用方式<Redirect to='路由位置'>,直要被渲染将直接跳转指定路由位置
              // 2. 实列( 当用户登陆后,则将无法访问注册/登陆页面 ):
          }
          <Route exact path='/sign' render={ ()=> this.props.currentUser ? <Redirect to='/' /> : <SignPage />  } />
          
          <Route exact path='/checkout' component={CheckoutPage} />
          <Route path='/tel' component={TelPage} />
        </Switch>
      </div>
    );
  }

}


// 测试路由 - 1 
// 关于Route标签传递的参数props( 完成笔记 )
  // 0. match属性对象:
    // a) url: 保存‘当前’在浏览器中的url链接( 仅限于保存符合路由规则的url,想要完整的url信息请看props.location.pathname属性解析 )
      // 0. 常用于创建动态路径( 具体看下方实战代码 )
    // b) path: 保存在Route标签中的path属性内容
      // 0. 如: <Route path='/pro/:proID' component={ProTextTest} /> 那么 props.match.path == '/pro/:proID';
    // c) isExact: 当url符合path规定的路由路径时为True, 否则为false
    // d) params: 常用于它的参数来配置'动态路由'页面跳转
      // 0. 常与Route标签中 path='/:xxx'下的':xxx'符号配合, 因为它能让:xxx变为params的属性
      // 1. 如: 
        /**
         * 路由配置: <Route path='/pro/:proID' component={ProTextTest} />
         * url访问: xxxx/pro/123
         * 那么params对象属性有: params = { proID: 123 }
         * **/
  // 1. history属性对象:
    // a) history下有很多函数属性,目前最常用的为push函数,用于跳转指定url页面,与Link标签功能相似
    // b) push(): 跳转指定url
      // 0. 如: <button onClick={ ()=> props.history.push('/pro') } > 产品页面 </button>
  // 2. location属性对象:
    // a) pathname: 保存完整的‘当前url信息’
      // 0. 如: 
        /**
         * 路由规则: <Route path='/pro/:proID' component={ProTextTest} />
         * 访问域名: http://localhost:3000/pro/123123/123123
         */
        // a) props.location.pathname = "/pro/123123/123123" ;
        // c) props.match.url = "/pro/123123";
        // b) 与match.url的区别: 在于pathname保留完整的url信息，但math.url只保存符合路由规则的url信息
// Link标签库：直接跳转到指定url( 完成笔记 )
  // 0. 如: <Link to="/" > 返回主页 </Link> 
  // 1. 在前端界面将被渲染为‘a标签’
/*
const HomePageTest = props => {
  console.log( '1',props );
  return ( 
    <div> 
      <h2 className="display-1" > 主页 </h2>
      <button onClick={ ()=> props.history.push('/pro') } > 产品页面 </button>
    </div> 
  );
}
const ProTest = props => {
  console.log( '2',props );
  return ( 
      <div> 
        <h2 className="display-1" > 产品页 </h2>
        <Link to="/" > 返回主页 </Link>      

        <Link to={ `${props.match.url}/13` } > 产品13 - match.url实战 </Link>      
        <Link to={ `${props.match.url}/14` } > 产品14 - match.url实战 </Link>      
      </div> 
    );
}
const ProTextTest = props => { 
  console.log( '3',props );
  return ( <div> <h2 className="display-1" > 产品详情页ID: { props.match.params.proID } </h2></div> ); 
}

function App() {
  return (
    <div className="App">
      <Switch>
        <Route exact path='/' component={HomePageTest} />
        <Route exact path='/pro' component={ProTest} />
        <Route path='/pro/:proID' component={ProTextTest} />
      </Switch>
    </div>
  );
}
*/
/*
// 测试路由 - 0
const HatsPage = () => ( <div> <h2 className="display-2" > TEST - Router </h2></div> );
const TestPage = () => ( <div> <h2 className="display-2" > TEST2 - Router </h2></div> );

function App() {
  return (
    // Route标签使用:( 完成笔记 )
      // 0. 模型: <Route exact={true/false} path="/" component={JSX} >默认3个参数
      // 1. exact属性:
        // a) exact为是否开启精确路由访问
        // b) 如果Route标签不写exact: 则默认exact为false否则为true
        // c) 关于exact属性的使用:
          // 0. 在<Switch>内，
            // a) 不开启exact: 则只能访问'/'路由页面
            // b) 开启exact: 则可以访问精确路由页面
          // 1. 不在<Switch>则进行叠加JSX渲染,就是根目录页面+指定路由页面渲染在同页
      // 2. path属性:
        // a) path="/"按照web路径来写
      // 3. component属性:
        // a) component={JSX}: 放置渲染的JSX变量
    <div className="App">
      <Switch>
        <Route exact={ true } path='/' component={HomePage} />
        <Route path='/hats' component={HatsPage} />
        <Route path='/test' component={TestPage} />
      </Switch>
    </div>
  );
}
*/
const mapStateToProps = createStructuredSelector ({
  currentUser: selectUserCurrentUser,
});

const mapDispatchToProps = dispatch => ({
  setCurrentUser: user => dispatch( setCurrentUser(user) ),
});

export default connect(mapStateToProps,mapDispatchToProps)(App);