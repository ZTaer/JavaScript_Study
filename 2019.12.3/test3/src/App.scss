@import "./_OO7BTS.v2.2";

body{
  @include font-style();
  @include choose-text-color();
}
.test{
  height:600px;
  background-image: linear-gradient(to right bottom, #FBDB89, #F48982);
  h1{
    color:linear-gradient(to right bottom, #FBDB89, #F48982);
  }
}
.test2{
  height:600px;

  background: #C6FFDD;  /* fallback for old browsers */
  background: -webkit-linear-gradient(to right, #f7797d, #FBD786, #C6FFDD);  /* Chrome 10-25, Safari 5.1-6 */
  background: linear-gradient(to right, #f7797d, #FBD786, #C6FFDD); /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */
}
.test3{
  height:600px;
  @include bg-linear( #93F9B9,#1D976C,right,bottom );
  @include bg-linear-yellow();
  @include bg-linear-dark();
  @include bg-linear-pink();
}