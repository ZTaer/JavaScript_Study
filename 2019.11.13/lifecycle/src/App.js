import React from 'react';
import logo from './logo.svg';
import './App.css';
import LifeCycle from './components/lifecycleTest/lifecycleTest.component';

class App extends React.Component{
  constructor(){
    super();
    this.state = {
      text: '',
      door: false,
    };

  }

  render(){
    return (
      <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />

        <button className="btn" onClick={ ()=>this.setState({ door: !this.state.door }) } >
            切换隐藏显示
        </button>
        <button className="btn" onClick={ ()=>this.setState({ text: this.state.text+'-hello-' }) } >
          增加-hello-数量
        </button>

        { this.state.door ? <LifeCycle state={ this.state } /> : null } 

      </header>
      </div>
    );
  }
}

export default App;
