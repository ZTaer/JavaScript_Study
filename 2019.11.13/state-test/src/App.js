import React from 'react';
import logo from './logo.svg';
import './App.css';

/**
 * class简洁版 - 依据ES6,class
 */
// 0. 无需constructor,因为他在class默认存在
// 1. 属性直接写即可，他会自动认定数据进入constructor下
// 2. 函数无变化
class App2 extends React.Component{
  state = {
    age: 12,
  }

  addAge = ()=>{
    this.setState( { age: this.state.age + 1 } );
  }

  render(){
    return ( 
        <div className="App2">
            <p>
              { this.state.age }
            </p>
            <button onClick={ this.addAge } >
              增加
            </button>
        </div>
     );
  }

}


/************* cut */

class App extends React.Component{
  constructor(){
    super();
    this.state = {
      age: 48,
    };

  }

  clickOnChangeAdd = () => {
    /**
     * this.setState()的本质
     */
    // 0. this.setState()为异步函数, 属于Component组件
    // 1. 本质用法: this.setState( ( prevState, prevProps )=>{ return {} }, ()=>{ '衔接异步数据处理' } );
        // a) 其实他有二个回调参数:
          // 0. prevState: 相当于复制this.state数据
          // 1. prevProps: 相当于复制this.props数据
          // 2. prevState/prevProps名称可变
        // b) 只所以异步是因为效率, React能根据最佳时机来改变this.state的数据
        // c) 正是因为有prevState,prevProps二个回调参数为对象类型, 他才诞生了‘正常用法必须使用对象的形式‘
        // d) 注意修改this.state数据: return { age: 'xxx' }依旧为对象类型
    // 2. 正常用法: this.setState( { xxx: 'yyy' }, ()=>{ '衔接异步数据处理' } );
    // 3. 关于this.props
        // a) this.props属于 Component组件的默认属性
        // b) 它包含的是'自定义标签'传递来的标签属性
            // 0. 如: < APP test="123" string="hahah" / > 那么this.props == {test:'123', string='hahah'};
    this.setState( 
      ( prevState, prevProps ) => { 
          console.log( prevState , prevProps ); // {age:48} , {test:123}
          return { age: prevState.age + prevProps.addNum  };
      }, 
      () => console.log( this.props ) // {test:123}
    );
    
  }

  render(){
    return ( 
        <div className="App">
            <p>
              { this.state.age }
            </p>
            <button onClick={ this.clickOnChangeAdd } >
              增加
            </button>
        </div>
     );
  }

/************* cut */

  /**
   * class中箭头函数和普通函数调用实战
  // 0. 箭头函数调用更加便利
  // 1. 普通函数调用较为麻烦

  // 0. 箭头函数
  clickOnChangeAdd = () => {
    this.setState({ age: this.state.age+1 });
  }

  // 1. 普通函数
  clickOnChangeLess(){
    this.setState({ age: this.state.age-1 });
  }

  render(){
    return ( 
        <div className="App">
          <header className="App-header">

            <p>
              { this.state.age }
            </p>

            // 箭头函数调用 
            <button onClick={ this.clickOnChangeAdd } >
              增加
            </button>

            // 普通函数调用方法
            <button onClick={ () => this.clickOnChangeLess() } >
              减少
            </button>
            
          </header>
        </div>
     );
  }
  */
}

export { App, App2 };
