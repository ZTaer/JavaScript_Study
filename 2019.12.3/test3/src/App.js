import React from 'react';
import logo from './logo.svg';
import './App.scss';

function App() {
  return (
    <div className="container-fluid">
      <div className="test">
        <h1 className="display-1 m-0 text-center py-5" >
          Test
        </h1>
      </div>
      <div className="test2" ></div>
      <div className="test3" ></div>
    </div>
  );
}

export default App;
