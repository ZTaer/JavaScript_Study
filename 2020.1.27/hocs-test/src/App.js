import React from 'react';
import './App.scss';

// import UserProfile from './component/user-profile/user-profile.component'; 
import UserProfile from './component-hocs/user-profile/user-profile.component'; 
import UserList from './component-hocs/user-list/user-list.component';

function App() {
  return (
    <div className="App">
      <UserProfile 
        name={`测试`} 
        email={`test@test.com`} 
        dataSource='https://jsonplaceholder.typicode.com/posts' 
      />
      <UserList 
        dataSource='https://jsonplaceholder.typicode.com/users' 
      />
    </div>
  );
}

export default App;
