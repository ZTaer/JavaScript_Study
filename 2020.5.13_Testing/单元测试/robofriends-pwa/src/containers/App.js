import React, { Component } from 'react';

import CardList from '../components/CardList';
import SearchBox from '../components/SearchBox';
import Scroll from '../components/Scroll';
import ErrorBoundry from '../components/ErrorBoundry';
import Header from '../components/Header';

import './App.css';

class App extends Component {
  constructor() {
    super()
    this.state = {
      count: 1
    }
  }

  componentDidMount() {
    this.props.onRequestRobots();
  }

  filteredRobots = () => {
    return this.props.robots ?
      this.props.robots.filter( robot => {
        return robot.name.toLowerCase().includes(this.props.searchField.toLowerCase());
    }) :
    []
  }

  render() {
    const { searchChange, isPending } = this.props;
    return (
      <div className='tc'>
        <Header count={this.state.count}/>

        <SearchBox searchChange={searchChange} />
        <Scroll>
          { isPending ? <h1>Loading</h1> :
            <ErrorBoundry>
              <CardList robots={this.filteredRobots()} />
            </ErrorBoundry>
          }
        </Scroll>
      </div>
    );
  }
}

export default App;
