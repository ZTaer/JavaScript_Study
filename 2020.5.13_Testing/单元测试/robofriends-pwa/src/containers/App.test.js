/**
 * 测试具有Redux的Class组件( 完成笔记 )
 */
// 0. 容器模式: 将app.js改进为容器模式，只测试jsx组件部分
// 1. beforeEach: 准备wrapper以及组件需要"参数模拟"保证运行正常
    // a) jest.fn(): JEST模拟函数
// 2. 快照: 将jsx进行快照备份

import { shallow } from 'enzyme';
import React from 'react';
import App from './App';

let wrapper;

// beforeEach: 在测试函数前运行, 为后续测试准备
beforeEach( ()=>{
    const mockProps = {
        onRequestRobots: jest.fn(),                 // JEST模拟函数( 完成笔记 )
        searchChange: '',
        isPending: false,
        robots: [],
    }
    wrapper = shallow(<App { ...mockProps } />);
} );

it('快照', () => {
    expect(wrapper).toMatchSnapshot();
});

// instance(): 获取class中的函数进行测试( 完成笔记 )
it('instance 测试class中的函数',()=>{
    expect( wrapper.instance().filteredRobots() ).toEqual([]);
});
