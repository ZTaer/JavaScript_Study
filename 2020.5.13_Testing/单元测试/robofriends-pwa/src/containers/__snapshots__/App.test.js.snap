// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`快照 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    isPending={false}
    onRequestRobots={[Function]}
    robots={Array []}
    searchChange=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header
          count={1}
/>,
        <SearchBox
          searchChange=""
/>,
        <Scroll>
          <ErrorBoundary>
                    <CardList
                              robots={Array []}
                    />
          </ErrorBoundary>
</Scroll>,
      ],
      "className": "tc",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "count": 1,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "searchChange": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ErrorBoundary>
            <CardList
                        robots={Array []}
            />
</ErrorBoundary>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <CardList
              robots={Array []}
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "robots": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header
            count={1}
/>,
          <SearchBox
            searchChange=""
/>,
          <Scroll>
            <ErrorBoundary>
                        <CardList
                                    robots={Array []}
                        />
            </ErrorBoundary>
</Scroll>,
        ],
        "className": "tc",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "count": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "searchChange": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ErrorBoundary>
              <CardList
                            robots={Array []}
              />
</ErrorBoundary>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <CardList
                robots={Array []}
/>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "robots": Array [],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
